src/WebJobs.Script/Workers/SharedMemoryDataTransfer/SharedMemoryManager.cs

```diff
@@ -0,0 +1,213 @@
+            {
+                return await PutBytesAsync(arr);
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        public void AddSharedMemoryMapForInvocation(string invocationId, string mapName)
+        {
+            IList<string> sharedMemoryMaps = _invocationSharedMemoryMaps.GetOrAdd(invocationId, new List<string>());
+            sharedMemoryMaps.Add(mapName);
+        }
+
+        /// <summary>
+        /// Reads data from the <see cref="MemoryMappedFile"/> with the given name.
+        /// Note:
+        /// Closes the reference to the <see cref="SharedMemoryMap"/> after reading.
+        /// </summary>
+        /// <param name="mapName">Name of the <see cref="MemoryMappedFile"/> to read from.</param>
+        /// <param name="offset">Offset to start reading data from in the
+        /// <see cref="MemoryMappedFile"/>.</param>
+        /// <param name="count">Number of bytes to read from, starting from the offset, in the
+        /// <see cref="MemoryMappedFile"/>.</param>
+        /// <returns>Data read as <see cref="byte[]"/> if successful, <see cref="null"/> otherwise.
+        /// </returns>
+        public async Task<byte[]> GetBytesAsync(string mapName, long offset, long count)
+        {
+            SharedMemoryMap sharedMemoryMap = Open(_logger, mapName);
+
+            try
+            {
+                byte[] content = await sharedMemoryMap.GetBytesAsync(offset, count);
+
+                if (content != null)
+                {
+                    return content;
+                }
+                else
+                {
+                    _logger.LogError($"Cannot read content from MemoryMappedFile: {mapName}");
+                    return null;
+                }
+            }
+            finally
+            {
+                sharedMemoryMap.DropReference();
+            }
+        }
+
+        public bool TryFreeSharedMemoryMapsForInvocation(string invocationId)
+        {
+            bool freedAll = true;
+            if (_invocationSharedMemoryMaps.TryRemove(invocationId, out IList<string> mapNames))
+            {
+                foreach (string mapName in mapNames)
+                {
+                    if (!TryFreeSharedMemoryMap(mapName))
+                    {
+                        freedAll = false;
+                    }
+                }
+            }
+
+            return freedAll;
+        }
+
+        public bool TryFreeSharedMemoryMap(string mapName)
+        {
+            if (_allocatedSharedMemoryMaps.TryRemove(mapName, out SharedMemoryMap sharedMemoryMap))
+            {
+                sharedMemoryMap.Dispose();
+                return true;
+            }
+            else
+            {
+                _logger.LogWarning($"Cannot find SharedMemoryMap: {mapName}");
+                return false;
+            }
+        }
+
+        public bool IsSupported(object input)
+        {
+            if (input is byte[] arr)
```

Comment:
Maybe out of scope, but I think a lot of folks would be happy to see this optimization on HTTP request bodies too (`input is HttpRequest request`)

Reply:
Good call out. Same here. For now scoping the change to binary data. Once we have initial implementation checked in and tested in prod - we can expand to HttpTriggers that send large data in the body - @gohar94 can you please open tracking issue for this.

Reply:
Done #6832